lib/rules/utils/ast-utils.js

```diff
@@ -1042,11 +1042,12 @@ function isStartOfExpressionStatement(node) {
 
 /**
  * Determines whether an opening parenthesis `(`, bracket `[` or backtick ``` ` ``` needs to be preceded by a semicolon.
- * This opening parenthesis or bracket should be at the start of an `ExpressionStatement` or at the start of the body of an `ArrowFunctionExpression`.
+ * This opening parenthesis or bracket should be at the start of an `ExpressionStatement`, a `MethodDefinition` or at
```

Comment:
In this case, suggestion adds unnecessary `;`:

```js
/* eslint require-await: "error" */

class C {
    foo() {}
    async [bar] () { baz; }
}
```

Reply:
Perhaps we could add `&& prevNode.parent.parent.type !== "MethodDefinition"` here to fix this:

https://github.com/eslint/eslint/blob/48117b27e98639ffe7e78a230bfad9a93039fb7f/lib/rules/utils/ast-utils.js#L1109

Reply:
👍 Nice. That fixes it and doesn't break anything else.